### PR TITLE
dateline wrap around

### PR DESCRIPTION
**Pull request by d-chambers**
Solves the issue of dateline wrap around for `get_events` and `EventBank` methods. See #230.

Hey @calum-chamberlain, here is the PR I promised. I didn't mean to steal your thunder, but as you can see, it required a good deal of refactoring some of the more complex parts of ObsPlus so I didn't want to subject you to that. Mind taking a look? A simple manual test on your data would also be good.

### CODE REVIEW

**Comment by calum-chamberlain** [comment]
Happy to have my thunder stolen! I will take a look. Thanks!

**Comment by calum-chamberlain** [comment]
Looks okay to me, a quick test on my NZ database with:
```python
df = bank.get_event_summary(
    starttime=UTCDateTime(2021, 3, 13), endtime=UTCDateTime(2021, 3, 15), 
    minlongitude=179, maxlongitude=-179)
```
returns a dataframe with longitudes spanning the 180 degree longitude, :+1:

However, running with:
```python
df = bank.get_event_summary(
    starttime=UTCDateTime(2021, 3, 13), endtime=UTCDateTime(2021, 3, 15), 
    minlongitude=179, maxlongitude=181)
```
doesn't return over the 180, and doesn't give any warning that 181 appears to be silently fudged to 180? Can this be fixed so that longitude > 180 is mapped nicely?

File: obsplus/events/get_events.py [hunk 1]
```diff
@@ -145,6 +146,22 @@ def _get_ids(df, kwargs) -> set:
...
 
 
+def _handle_dateline_transversal(filt, df, kwargs):
+    """Check if dateline should be transversed by query."""
+    # if longitudes aren't being used bail out
+    if not {"minlongitude", "maxlongitude"}.issubset(set(kwargs)):
+        return filt, kwargs
+    # if dateline is not to be transversed by query bail out
+    minlong, maxlong = kwargs["minlongitude"], kwargs["maxlongitude"]
+    if not minlong > maxlong:
```
Comment (calum-chamberlain) [inline obsplus/events/get_events.py:156]: I think when `maxlongitude` is given incorrectly as > 180 this ignores the dateline issue.

File: obsplus/events/get_events.py [hunk 2]
```diff
@@ -145,6 +146,22 @@ def _get_ids(df, kwargs) -> set:
     return set(df.event_id[:limit])
 
 
+def _handle_dateline_transversal(filt, df, kwargs):
+    """Check if dateline should be transversed by query."""
+    # if longitudes aren't being used bail out
+    if not {"minlongitude", "maxlongitude"}.issubset(set(kwargs)):
+        return filt, kwargs
+    # if dateline is not to be transversed by query bail out
+    minlong, maxlong = kwargs["minlongitude"], kwargs["maxlongitude"]
```
Comment (calum-chamberlain) [inline obsplus/events/get_events.py:155]:
```suggestion
    minlong, maxlong = kwargs["minlongitude"], kwargs["maxlongitude"]
    minlong = ((minlong + 180) % 360) - 180
    maxlong = ((maxlong + 180) % 360) - 180
```
this should help cope with people (like me) doing dumb things?

File: obsplus/events/get_events.py (hunk 1 above)
Comment (d-chambers) [reply obsplus/events/get_events.py:157]: You are right, I forced the longitude values to be in the expected range in other parts of the code but not for the input to these queries. Luckily I made a nice little utility for this already, so it was an easy fix. It should work as expected now.

**Comment by d-chambers** [comment]
Ok, this PR should solve your problem @calum-chamberlain. If not, we can open another issue.

**Pull request merged**